### PR TITLE
[Dev Portal] [Application Management] - Application template selection page.

### LINKED ISSUE #410: [Dev Portal] [Application Management] - UI breaks when the`advancedConfigurations` attribute in application GET is undefined
**Describe the bug**
UI breaks when the user navigates to the settings tab of an application which hasn't configured and `advancedConfigurations`.

**To Reproduce**
1. Go to the applications page in the developer portal.
2. Click on an application which hasn't configured `advancedConfigurations`
4. Check the console for the stack trace.

**Expected behavior**
UI shouldn't break and the user should be able to configure the `advancedConfigurations`.

**Screenshots**
Screen Shot 2020-02-13 at 7 09 08 PM

**Desktop (please complete the following information):**
- OS: MacOS Mojave 10.14.6
- Browser: Chrome
- Version: 79.0.3945.130

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose
Users should be able to select one of the pre configured application templates to initiate the application creation wizard.

## Goals
This PR contains an improved version of the already existing template selection page.
Also Fixes #410.

## Approach

Screen Shot 2020-02-13 at 7 22 44 PM

## TODO
1. Add localization support.

### CODE REVIEW

File: modules/theme/src/themes/default/assets/images/illustrations/spa-illustration.svg [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/illustrations/spa-illustration.svg]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/theme/src/themes/default/assets/images/illustrations/web-app-illustration.svg [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/illustrations/web-app-illustration.svg]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/theme/src/themes/default/assets/images/technologies/java-logo.svg [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/technologies/java-logo.svg]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/theme/src/themes/default/assets/images/technologies/react-logo.svg [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/technologies/react-logo.svg]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/theme/src/themes/default/assets/images/technologies/vue-logo.svg [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/technologies/vue-logo.svg]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+-->
+
+<svg version="1.1" id="angular-logo" class="icon" width="14" height="14" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 250 250" xml:space="preserve">
+	<style type="text/css">
+		.st0{fill:#DD0031;}
+		.st1{fill:#C3002F;}
+		.st2{fill:#FFFFFF;}
+	</style>
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg]: Shall we have these inline

Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg]: Fixed with commit `3246462`

File: modules/theme/src/themes/default/assets/images/technologies/vue-logo.svg (hunk 6 above)
Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/technologies/vue-logo.svg]: Fixed with commit `ed4adf0`

File: modules/theme/src/themes/default/assets/images/technologies/react-logo.svg (hunk 5 above)
Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/technologies/react-logo.svg]: Fixed with commit `ed4adf0`

File: modules/theme/src/themes/default/assets/images/technologies/java-logo.svg (hunk 4 above)
Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/technologies/java-logo.svg]: Fixed with commit `ed4adf0`

File: modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg (hunk 3 above)
Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/technologies/angular-logo.svg]: Fixed with commit `ed4adf0`

File: modules/theme/src/themes/default/assets/images/illustrations/web-app-illustration.svg (hunk 2 above)
Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/illustrations/web-app-illustration.svg]: Fixed with commit `ed4adf0`

File: modules/theme/src/themes/default/assets/images/illustrations/spa-illustration.svg (hunk 1 above)
Comment (brionmario) [reply modules/theme/src/themes/default/assets/images/illustrations/spa-illustration.svg]: Fixed with commit `ed4adf0`

**Pull request merged**